zipline/pipeline/loaders/events.py

```diff
@@ -36,196 +42,135 @@
-                    v.index = [dates[0]] * len(v)
-                else:
-                    self.events_by_sid[k] = v.set_index(TS_FIELD_NAME)
-                # Once data is in a DF, make sure columns are correct.
-                cols_except_ts = (set(v.columns) -
-                                  {TS_FIELD_NAME})
-
-                # Check that all columns other than timestamp are as expected.
-                if cols_except_ts != self.expected_cols:
-                    raise ValueError(
-                        WRONG_COLS_ERROR.format(
-                            expected_columns=list(self.expected_cols),
-                            sid=k,
-                            resulting_columns=v.columns.values
-                        )
-                    )
-            # Not a DataFrame and we only expect 1 column
-            elif len(self.expected_cols) == 1:
-                # First, must convert to DataFrame.
-                if isinstance(v, pd.Series):
-                    if not isinstance(v.index, pd.DatetimeIndex):
-                        raise ValueError(
-                            SERIES_NO_DTINDEX_ERROR.format(sid=k)
-                        )
-                    self.events_by_sid[k] = pd.DataFrame({
-                        list(self.expected_cols)[0]: v})
-                elif isinstance(v, pd.DatetimeIndex):
-                    if not infer_timestamps:
-                        raise ValueError(
-                            DTINDEX_NOT_INFER_TS_ERROR.format(sid=k)
-                        )
-                    self.events_by_sid[k] = pd.DataFrame({
-                        list(self.expected_cols)[0]: v
-                    }, index=[dates[0]] * len(v))
-                else:
-                    # We expect 1 column, but we got something other than a
-                    # Series, DatetimeIndex, or DataFrame.
-                    raise ValueError(
-                        WRONG_SINGLE_COL_DATA_FORMAT_ERROR.format(sid=k)
-                    )
-            else:
-                # We expected multiple columns, but we got something other
-                # than a DataFrame.
-                raise ValueError(
-                    WRONG_MANY_COL_DATA_FORMAT_ERROR.format(sid=k)
-                )
-        self.events_by_sid = {sid: df.dropna(subset=[self.event_date_col]) for
-                              sid, df in self.events_by_sid.items()}
-        self.dataset = dataset
-
-    def get_loader(self, column):
-        if column in self.dataset.columns:
-            return getattr(self, "%s_loader" % column.name)
-        raise ValueError("Don't know how to load column '%s'." % column)
-
-    def load_adjusted_array(self, columns, dates, assets, mask):
-        return merge(
-            self.get_loader(column).load_adjusted_array(
-                [column], dates, assets, mask
+                 events,
+                 next_value_columns,
+                 previous_value_columns):
```

Comment:
Maybe we should assert on construction of `EventsLoader` that the columns in  `next_value_columns` and `previous_value_columns` are contained in  `events` rather than waiting to break when we try to load the adjusted array.